src/zenml/zen_stores/base_zen_store.py

```diff
@@ -78,6 +80,45 @@ class BaseZenStore(
     TYPE: ClassVar[StoreType]
     CONFIG_TYPE: ClassVar[Type[StoreConfiguration]]
 
+    @model_validator(mode="before")
+    @classmethod
+    @before_validator_handler
+    def convert_config(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Method to infer the correct type of the config and convert.
+
+        Args:
+            data: The provided configuration object, can potentially be a
+                generic object
+
+        Raises:
+            ValueError: If the provided config object's type does not match
+                any of the current implementations.
+
+        Returns:
+            The converted configuration object.
+        """
+        if data["config"].type == StoreType.SQL:
+            from zenml.zen_stores.sql_zen_store import SqlZenStoreConfiguration
+
+            data["config"] = SqlZenStoreConfiguration(
+                **data["config"].model_dump()
+            )
```

Comment:
Cannot we just `cast` here? Doesn't it work as expected?
Also why do you need this at all, if the class is an instance of a child, why is it bad just to use it as-is? Like we did before, I  assume.

Reply:
Unfortunately, we can not use the `cast` here. Preivously this would have worked, but with pydantic v2, when you instantiate a model, you need to provide either a dict or an instance of the same class. Any instances of subclasses do not work and throw a ValidationError.